pkg/controller/v1alpha1/inferencegraph/knative_reconciler.go

```diff
@@ -274,3 +273,67 @@ func constructResourceRequirements(graph v1alpha1api.InferenceGraph, config Rout
+		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
+	}
+
+	// If a minReplicas value is not set for the inference graph, then use the default min-scale value of 1.
+	var revisionMinScale int
+	if _, ok := annotations[autoscaling.MinScaleAnnotationKey]; !ok {
+		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(constants.DefaultMinReplicas)
+		revisionMinScale = constants.DefaultMinReplicas
+	} else {
+		log.Info("inference graph contains min-scale annotation. This value will be used instead of the configured minReplicas",
+			"min-scale", annotations[autoscaling.MinScaleAnnotationKey])
+		minScaleInt, err := strconv.Atoi(annotations[autoscaling.MinScaleAnnotationKey])
+		if err != nil {
+			return errors.Wrapf(err, "fails to convert existing min-scale annotation to an integer")
+		}
+		revisionMinScale = minScaleInt
+	}
+
+	// Retrieve the allow-zero-initial-scale and initial-scale values from the knative autoscaler configuration.
+	allowZeroInitialScale, globalInitialScale, err := knutils.GetAutoscalerConfiguration(client)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve the knative autoscaler configuration")
+	}
+
+	initialScaleInt, err := strconv.Atoi(globalInitialScale)
+	if err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("fails to convert configured knative serving global initial-scale value to an integer. initial-scale: %s", globalInitialScale))
+	}
+
+	// Provide transparency to users while aligning with knative serving's expected behavior, log a warning when the
+	// knative autoscaler's gloabal initial-scale value exceeds the requested minScale value for the inference graph.
+	if initialScaleInt > revisionMinScale {
+		log.Info("WARNING: knative autoscaler is globally configured with an initial-scale value that is greater than the requested min-scale for the inference graph",
+			"initial-scale", initialScaleInt,
+			"min-scale", revisionMinScale)
+	}
+
+	// knative will choose the larger of min-scale and initial-scale as the initial target scale for a knative revision.
+	// When min-scale is 0, if allow-zero-initial scale is true, set initial-scale to 0 for the created knative revision.
+	// This will prevent any pods from being created to initialize a knative revision when an inference graph has minReplicas set to 0.
+	// Configuring scaling for knative: https://knative.dev/docs/serving/autoscaling/scale-bounds/#initial-scale
+	if revisionMinScale == 0 {
+		if allowZeroInitialScale == "true" {
+			log.Info("kserve will override the global knative autoscaler configuration for initial-scale on a per revision basis with 0 when an inference graph is requested with min-scale 0",
+				"allow-zero-initial-scale", allowZeroInitialScale,
+				"initial-scale", globalInitialScale)
+			annotations[constants.InitialScaleAnnotationKey] = "0"
+		} else {
+			log.Info("WARNING: The current knative autoscaler global configuration does not allow zero initial scale.",
```

Comment:
do we need to add `WARNING` in the log?  Why don't we update the message to reflect the warning context?
```
log.Info("The current knative autoscaler global configuration does not allow zero initial scale, which may lead to scaling issues.", ...
```